text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,399 @@
+|🫣 | `Ember.__loader.registry` | [13 addons, ~5 recent](https://emberobserver.com/code-search?codeQuery=Ember.__loader.registry&sort=updated&sortAscending=false). One is `ember-cli-fastboot` (tests, test-support). | n/a |
+|🔒 | `Ember.BOOTED` | [None](https://emberobserver.com/code-search?codeQuery=Ember.BOOTED&sort=updated&sortAscending=false) | n/a |
+
+
+Replaced by [RFC #931][RFC-931].
+For scenarios where folks would like to compile templates at runtime, see [RFC #931][RFC-931] or the code of [ember-repl](https://www.npmjs.com/package/ember-repl).
+|   | API | Usage: EmberObserver | Migration |
+| - | --- | ----- | --------- |
+|🔒 | `Ember.TEMPLATES` | [`ember-resolver`](https://emberobserver.com/code-search?codeQuery=Ember.TEMPLATES&sort=updated&sortAscending=false) | n/a |
+|🫣 | `Ember.HTMLBars` | [Lots of usage (encompasses the below APIs)](https://emberobserver.com/code-search?codeQuery=Ember.HTMLBars&sort=updated&sortAscending=false) | n/a |
+|🫣 | `Ember.HTMLBars.DOMHelper` | [`ember-cli-fastboot`](https://emberobserver.com/code-search?codeQuery=Ember.HTMLBars.DOMHelper&sort=updated&sortAscending=false) uses `protocolForURL`, `parseHTML` | n/a |
+|🫣 | `Ember.HTMLBars.template` | [`ember-cli-fastboot`](https://emberobserver.com/code-search?codeQuery=Ember.HTMLBars.template&sort=updated&sortAscending=false) (and `ember-ast-hot-load`) | n/a |
+|🫣 | `Ember.HTMLBars.compile` | [old addons](https://emberobserver.com/code-search?codeQuery=Ember.HTMLBars.compile&sort=updated&sortAscending=false) | n/a |
+|🫣 | `Ember.HTMLBars.precomple` | [`ember-ast-hot-load`](https://emberobserver.com/code-search?codeQuery=Ember.HTMLBars.precompile&sort=updated&sortAscending=false) | n/a |
+|🫣 | `Ember.Handlebars` | [174 addons, mostly @ascua](https://emberobserver.com/code-search?codeQuery=Ember.Handlebars&sort=updated&sortAscending=false) and also a lot of mentions in docs. | n/a |
+|🫣 | `Ember.Handlebars.template` | [None](https://emberobserver.com/code-search?codeQuery=Ember.Handlebars.template&sort=updated&sortAscending=false) | n/a |
+|🫣 | `Ember.Handlebars.Utils.escapeExpression` | [100 addons, mostly @ascua](https://emberobserver.com/code-search?codeQuery=Ember.Handlebars.Utils.escapeExpression&sort=updated&sortAscending=false) | Removed in [ember.js PR#20360](https://github.com/emberjs/ember.js/pull/20360) as it is not public API. |
+|🫣 | `Ember.Handlebars.compile` | [`ember-cli-fastboot` and `ember-collection`](https://emberobserver.com/code-search?codeQuery=Ember.Handlebars.compile&sort=updated&sortAscending=false) | n/a |
+|🫣 | `Ember.Handlebars.precomple` | [None](https://emberobserver.com/code-search?codeQuery=Ember.Handlebars.precompile&sort=updated&sortAscending=false) | n/a |
+
+
+Other APIs
+- 🫣 `Ember.testing`  
+  Instead, use
+
+  ```js
+  import { macroCondition, isTesting } from '@embroider/macros';
+
+  // ...
+
+  if (macroCondition(isTesting())) {
+    // test only code here
+  }
+  ```
+
+- 🌐 `Ember.onerror`
```

Comment:
It would be updated to use the error event on the window 🎉

Reply:
If you define an error event handler it doesn't allow you to catch the error and the error gets propagated up to the testem's [global error handler](https://github.com/testem/testem/blob/c055067ab88d6061ebd3ea19b2eff39c5428891f/public/testem/testem_client.js#L414). You can't always surround the test with a try/catch block, specifically if the rejection happens inside an ember-concurrency task.

Reply:
An alternate route we can use is to have the ember handler exported explicitly.

This is basically the strategy for the whole RFC, if an api is actually needed somewhere, an import must exist for it, as this RFC's primary goal is to get rid of the barrel file 'ember', rather than to get rid of features

Reply:
Absolutely! Just wanted to highlight that Ember.onerror needs further investigation.

Reply:
Maybe we can stub `window.onerror` instead of using setupOnerror? Just confirmed that we prevent testem's error handler by calling `window.onerror = () => true;` in the test. Of course, we would want to restore the original value after the expected exception is handled.

Reply:
that's basically how `setupOnerror` and `resetOnerror` work :tada:

Reply:
They do it with `Ember.onerror` and in the new world without it, they would need to do it with `window.onerror`.